src/app/terminal.js

```diff
@@ -284,39 +307,46 @@ class Terminal {
             // Received RTS
             case Broadcast.types.RTS: {
                 if (receivedBroadcast.destination === this && !this.sentData) {
+                    this.rtsTimeAccumulator=0;
+                    this.nextRtsTime+=receivedBroadcast.data[1];
                     this.queueBroadcast(Broadcast.types.CTS, receivedBroadcast.source, [receivedBroadcast.id,receivedBroadcast.data]);
                 }
                 break;
             }
 
             // Received CTS
             case Broadcast.types.CTS: {
+                this.rtsTimeAccumulator=0;
+                this.nextRtsTime+=receivedBroadcast.data[1];
                 if (receivedBroadcast.destination === this) {
                     const index = this.unackedRtses.findIndex(id => receivedBroadcast.data[0] === id);
                     if (index !== -1) {
                         this.unackedRtses.splice(index, 1);
+                        this.dataTimeout=receivedBroadcast.data[1]/numberOfAttempts;
```

Comment:
What if `numberOfAttempts=0`?

Reply:
it's a const it's max number of attempts*